### PR TITLE
planner: rename `general plan cache` to `non-prepared plan cache`

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: close #xxx

Problem Summary: planner: rename `general plan cache` to `non-prepared plan cache`

### What is changed and how it works?

No logical change, just refactor. planner: rename `general plan cache` to `non-prepared plan cache`

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: planner/core/plan_cache_utils.go [hunk 1]
```diff
@@ -161,21 +161,6 @@ func GeneratePlanCacheStmtWithAST(ctx context.Context, sctx sessionctx.Context,
 	return preparedObj, p, ParamCount, nil
 }
 
-func getValidPlanFromCache(sctx sessionctx.Context, isGeneralPlanCache bool, key kvcache.Key, paramTypes []*types.FieldType) (*PlanCacheValue, bool) {
```
Comment (qw4990) [inline planner/core/plan_cache_utils.go:164]: useless functions

File: planner/optimize.go [hunk 2]
```diff
@@ -436,7 +436,7 @@ func OptimizeExecStmt(ctx context.Context, sctx sessionctx.Context,
 	if !ok {
 		return nil, nil, errors.Errorf("invalid result plan type, should be Execute")
 	}
-	plan, names, err := core.GetPlanFromSessionPlanCache(ctx, sctx, execAst.FromGeneralStmt, is, exec.PrepStmt, exec.Params)
+	plan, names, err := core.GetPlanFromSessionPlanCache(ctx, sctx, false, is, exec.PrepStmt, exec.Params)
```
Comment (qw4990) [inline planner/optimize.go:439]: Always 'false' here.

File: sessionctx/variable/sysvar.go [hunk 3]
```diff
@@ -1063,13 +1063,13 @@ var defaultSysVars = []*SysVar{
 		return strconv.FormatFloat(PreparedPlanCacheMemoryGuardRatio.Load(), 'f', -1, 64), nil
 	}},
 	{Scope: ScopeGlobal | ScopeSession, Name: TiDBEnableGeneralPlanCache, Value: BoolToOnOff(DefTiDBEnableGeneralPlanCache), Type: TypeBool, SetSession: func(s *SessionVars, val string) error {
```
Comment (qw4990) [inline sessionctx/variable/sysvar.go:1065]: I'll update these variable names in the next PR.

File: executor/prepared.go [hunk 4]
```diff
@@ -145,7 +145,7 @@ func (e *PrepareExec) Next(ctx context.Context, req *chunk.Chunk) error {
 	e.ParamCount = paramCnt
 	e.Stmt = stmt
 	if e.IsGeneralStmt {
```
Comment (fzzf678) [inline executor/prepared.go:147]: should  the `IsGeneralStmt`  also be renamed？

Comment (qw4990) [reply executor/prepared.go]: Updated, PTAL

**Review by fixdb — APPROVED** [review]
+1

**Review by fzzf678 — APPROVED** [review]

**Review by XuHuaiyu — APPROVED** [review]

**Pull request merged**